code/_onclick/hud/parallax.dm

```diff
@@ -0,0 +1,300 @@
+	C.previous_turf = posobj
+	C.last_parallax_shift = world.time
+
+	for(var/thing in C.parallax_layers)
+		var/obj/screen/parallax_layer/L = thing
+		L.update_status(mymob)
+		if(L.view_sized != C.view)
+			L.update_o(C.view)
+		
+		var/change_x
+		var/change_y
+
+		if(L.absolute)
+			L.offset_x = -(posobj.x - SSparallax.planet_x_offset) * L.speed
+			L.offset_y = -(posobj.y - SSparallax.planet_y_offset) * L.speed
+		else
+			change_x = offset_x * L.speed
+			L.offset_x -= change_x
+			change_y = offset_y * L.speed
+			L.offset_y -= change_y
+
+			if(L.offset_x > 240)
+				L.offset_x -= 480
+			if(L.offset_x < -240)
+				L.offset_x += 480
+			if(L.offset_y > 240)
+				L.offset_y -= 480
+			if(L.offset_y < -240)
+				L.offset_y += 480
+
+		if(!areaobj.parallax_movedir && C.dont_animate_parallax <= world.time && (offset_x || offset_y) && abs(offset_x) <= max(C.parallax_throttle/world.tick_lag+1,1) && abs(offset_y) <= max(C.parallax_throttle/world.tick_lag+1,1) && (round(abs(change_x)) > 1 || round(abs(change_y)) > 1))
+			L.transform = matrix(1, 0, offset_x*L.speed, 0, 1, offset_y*L.speed)
+			animate(L, transform=matrix(), time = last_delay)
+
+		L.screen_loc = "CENTER-7:[round(L.offset_x,1)],CENTER-7:[round(L.offset_y,1)]"
+
+/atom/movable/proc/update_parallax_contents()
+	if(length(client_mobs_in_contents))
+		for(var/thing in client_mobs_in_contents)
+			var/mob/M = thing
+			if(M && M.client && M.hud_used && length(M.client.parallax_layers))
+				M.hud_used.update_parallax()
+
+/obj/screen/parallax_layer
+	icon = 'icons/effects/parallax.dmi'
+	var/speed = 1
+	var/offset_x = 0
+	var/offset_y = 0
+	var/view_sized
+	var/absolute = FALSE
+	blend_mode = BLEND_ADD
+	plane = PLANE_SPACE_PARALLAX
+	screen_loc = "CENTER-7,CENTER-7"
+	mouse_opacity = 0
+
+
+/obj/screen/parallax_layer/New(view)
+	..()
+	if(!view)
+		view = world.view
+	update_o(view)
+
+/obj/screen/parallax_layer/proc/update_o(view)
+	if(!view)
+		view = world.view
+	var/list/new_overlays = list()
+	var/count = Ceiling(view/(480/world.icon_size))+1
+	for(var/x in -count to count)
+		for(var/y in -count to count)
+			if(x == 0 && y == 0)
+				continue
+			var/image/I = image(icon, null, icon_state)
+			I.transform = matrix(1, 0, x*480, 0, 1, y*480)
+			new_overlays += I
+
+	overlays = new_overlays
+	view_sized = view
+
+/obj/screen/parallax_layer/proc/update_status(mob/M)
+	return
+
+/obj/screen/parallax_layer/layer_1
+	icon_state = "layer1"
+	speed = 0.6
+	layer = 1255
+
+/obj/screen/parallax_layer/layer_2
+	icon_state = "layer2"
+	speed = 1
+	layer = 2
+
+/obj/screen/parallax_layer/planet
+	icon_state = "planet"
+	blend_mode = BLEND_OVERLAY
+	absolute = TRUE //Status of seperation
+	speed = 3
+	layer = 30
+
+/obj/screen/parallax_layer/planet/update_status(mob/M)
+	var/turf/T = get_turf(M)
+	if(is_station_level(T.z))
+		invisibility = 0
+	else
+		invisibility = INVISIBILITY_ABSTRACT
+
+/obj/screen/parallax_layer/planet/update_o()
+	return //Shit wont move
+
+/obj/screen/parallax_pmaster
```

Comment:
Not sure the purpose of this one.

Reply:
It looks like @AffectedArc07 refactored some of the code in `create_parallax` into layers that get added in `parallax_static_layers_tail` instead (`parallax_pmaster` and `parallax_space_whitifier`. Not sure what the difference is between the two approaches